llvm/test/CodeGen/AIE/aie2/GlobalISel/prelegalizercombiner-s20-narrowing.mir

```diff
@@ -781,3 +781,294 @@ body:             |
+    ; CHECK-NEXT: $p0 = COPY [[INT]](p0)
+    %0:_(s32) = G_CONSTANT i32 7
+    %1:_(s32) = G_CONSTANT i32 0
+    %2:_(<16 x s32>) = COPY $x0
+    %3:_(s32) = G_INTRINSIC intrinsic(@llvm.aie2.vextract.elem32.I512), %2(<16 x s32>), %0(s32), %1(s32)
+    %4:_(s20) = G_TRUNC %3(s32)
+    %5:_(p0) = G_CONSTANT i20 0
+    %6:_(p0), %7:_(s20) = G_INTRINSIC intrinsic(@llvm.aie2.add.2d), %5:_(p0), %4:_(s20), %4:_(s20), %4:_(s20), %4:_(s20)
+    $p0 = COPY %6
+...
+
+# Only one Src Node (vextract8) for G_PTR_ADD that is narrowed to S20 type, intermediate G_TRUNC is removed
+---
+name:            valid_vextract8_PTR_ADD
+legalized:       false
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LABEL: name: valid_vextract8_PTR_ADD
+  ; CHECK: bb.0:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT:   liveins: $p0, $x0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[COPY:%[0-9]+]]:_(p0) = COPY $p0
+  ; CHECK-NEXT:   [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 0
+  ; CHECK-NEXT:   [[COPY1:%[0-9]+]]:_(<64 x s8>) = COPY $x0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[AIE_SEXT_EXTRACT_VECTOR_ELT:%[0-9]+]]:_(s20) = G_AIE_SEXT_EXTRACT_VECTOR_ELT [[COPY1]](<64 x s8>), [[C]](s32)
+  ; CHECK-NEXT:   [[PTR_ADD:%[0-9]+]]:_(p0) = G_PTR_ADD [[COPY]], [[AIE_SEXT_EXTRACT_VECTOR_ELT]](s20)
+  ; CHECK-NEXT:   G_STORE [[C]](s32), [[PTR_ADD]](p0) :: (store (s32))
+  ; CHECK-NEXT:   G_BR %bb.1
+  bb.1:
+    successors: %bb.2(0x80000000); %bb.2(100.00%)
+    liveins: $p0, $x0
+    %0:_(p0) = COPY $p0
+    %1:_(s32) = G_CONSTANT i32 0
+    %2:_(s32) = G_CONSTANT i32 1
+    %3:_(<64 x s8>) = COPY $x0
+    %4:_(s32) = G_INTRINSIC intrinsic(@llvm.aie2.vextract.elem8.I512), %3(<64 x s8>), %1(s32), %2(s32)
+
+  bb.2:
+    successors: %bb.2(0x80000000); %bb.2(100.00%)
+
+    %5:_(s20) = G_TRUNC %4:_(s32)
+    %6:_(p0) = G_PTR_ADD %0:_, %5:_(s20)
+    G_STORE %1:_(s32), %6:_(p0) :: (store (s32))
+    G_BR %bb.2
+...
+
+# Only one Src Node (vextract16) for G_PTR_ADD that is narrowed to S20 type, intermediate G_TRUNC is removed
+---
+name:            valid_vextract16_PTR_ADD
+legalized:       false
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LABEL: name: valid_vextract16_PTR_ADD
+  ; CHECK: bb.0:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT:   liveins: $p0, $x0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[COPY:%[0-9]+]]:_(p0) = COPY $p0
+  ; CHECK-NEXT:   [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 0
+  ; CHECK-NEXT:   [[COPY1:%[0-9]+]]:_(<32 x s16>) = COPY $x0
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT: bb.1:
+  ; CHECK-NEXT:   successors: %bb.1(0x80000000)
+  ; CHECK-NEXT: {{  $}}
+  ; CHECK-NEXT:   [[AIE_SEXT_EXTRACT_VECTOR_ELT:%[0-9]+]]:_(s20) = G_AIE_SEXT_EXTRACT_VECTOR_ELT [[COPY1]](<32 x s16>), [[C]](s32)
```

Comment:
same issue as mentioned in https://github.com/Xilinx/llvm-aie/pull/233/files#r1834826147